components/apimgt/org.wso2.carbon.apimgt.rest.api.publisher.v1.common/src/main/java/org/wso2/carbon/apimgt/rest/api/publisher/v1/common/mappings/PublisherCommonUtils.java

```diff
@@ -874,19 +875,37 @@ public static boolean validateEndpoints(APIDTO apiDto) {
             return true;
         }
 
-        // extract sandbox endpoint URL
-        if (!endpointConfiguration.isNull(APIConstants.API_DATA_SANDBOX_ENDPOINTS)) {
-            endpoints.add(endpointConfiguration.getJSONObject(APIConstants.API_DATA_SANDBOX_ENDPOINTS)
-                    .getString(APIConstants.API_DATA_URL));
-        }
-        // extract production endpoint URL
-        if (!endpointConfiguration.isNull(APIConstants.API_DATA_PRODUCTION_ENDPOINTS)) {
-            endpoints.add(endpointConfiguration.getJSONObject(APIConstants.API_DATA_PRODUCTION_ENDPOINTS)
-                    .getString(APIConstants.API_DATA_URL));
-        }
+        // extract sandbox endpoint URL(s)
+        extractURLsFromEndpointConfig(endpointConfiguration, APIConstants.API_DATA_SANDBOX_ENDPOINTS, endpoints);
+
+        // extract production endpoint URL(s)
+        extractURLsFromEndpointConfig(endpointConfiguration, APIConstants.API_DATA_PRODUCTION_ENDPOINTS, endpoints);
+
         return APIUtil.validateEndpointURLs(endpoints);
     }
 
+    /**
+     * Extract sandbox or production endpoint URLs from endpoint config object.
+     *
+     * @param endpointConfigObj Endpoint config JSON object
+     * @param endpointType      Indicating whether Sandbox or Production endpoints are to be extracted
+     * @param endpoints         List of URLs. Extracted URL(s), if any, are added to this list.
+     */
+    private static void extractURLsFromEndpointConfig(org.json.JSONObject endpointConfigObj, String endpointType,
+            ArrayList<String> endpoints) {
+        if (!endpointConfigObj.isNull(endpointType)) {
+            org.json.JSONObject endpointObj = endpointConfigObj.optJSONObject(endpointType);
+            if (endpointObj != null) {
+                endpoints.add(endpointConfigObj.getJSONObject(endpointType).getString(APIConstants.API_DATA_URL));
+            } else {
+                JSONArray endpointArray = endpointConfigObj.getJSONArray(endpointType);
+                for (int i = 0; i < endpointArray.length(); i++) {
```

Comment:
Shall we switch to a forEach loop ?

Reply:
Seems like we can't iterate through JSONArray with a forEach